src/SpectatorView.Unity/Assets/SpectatorView.Editor/Shaders/Blur.shader

```diff
@@ -3,83 +3,145 @@
     Properties
     {
         _MaskTexture("MaskTexture", 2D) = "white" {}
-        _BlurIntensity("BlurIntensity", float) = 0.00375
+        _BlurSize("BlurSize", Range(0, 10)) = 5
     }
-    SubShader
-    {
-        // No culling or depth
-        Cull Off ZWrite Off ZTest Always
 
-        Pass
-        {
-            CGPROGRAM
-            #pragma vertex vert
-            #pragma fragment frag
+    Category
+     {
+            // We must be transparent, so other objects are drawn before this one.
+            Tags { "Queue" = "Transparent" "IgnoreProjector" = "True" "RenderType" = "Opaque" }
 
-            #include "UnityCG.cginc"
 
-            struct appdata
+        SubShader
+        {
+            // No culling or depth
+            Cull Off ZWrite Off ZTest Always
+            // Horizontal blur
+            Pass
             {
-                float4 vertex : POSITION;
-                float2 uv : TEXCOORD0;
-            };
+                CGPROGRAM
+                #pragma vertex vert
+                #pragma fragment frag
 
-            struct v2f
-            {
-                float2 uv : TEXCOORD0;
-                float4 vertex : SV_POSITION;
-            };
+                #include "UnityCG.cginc"
+
+                struct appdata
+                {
+                    float4 vertex : POSITION;
+                    float2 uv : TEXCOORD0;
+                };
+
+                struct v2f
+                {
+                    float2 uv : TEXCOORD0;
+                    float4 vertex : SV_POSITION;
+                };
+
+                v2f vert(appdata v)
+                {
+                    v2f o;
+                    o.vertex = UnityObjectToClipPos(v.vertex);
+                    o.uv = v.uv;
+                    return o;
+                }
 
-            v2f vert (appdata v)
+                float4 _MaskTexture_TexelSize;
+                Texture2D _MaskTexture;
+                SamplerState sampler_point_clamp;
+                float _BlurSize;
+
+                fixed4 frag(v2f i) : SV_Target
+                {
+                    half4 sum = half4(0,0,0,0);
+                    float sumX = 0.0;
+
+                    #define MASKSAMPLE(weight, kernalx) _MaskTexture.Sample(sampler_point_clamp, float2(i.uv[0] - kernalx * _BlurSize * _MaskTexture_TexelSize.x, i.uv[1])).r * weight;
```

Comment:
>kernalx [](start = 47, length = 7)

Is "kernel" the right term here? I always thought it was spelled that way but now I'm questioning it

Reply:
Just a typo! Fixed